### PR TITLE
Config Filters; Sentry; extra props here and there

### PR DESCRIPTION
**Pull request by SandyRogers**
This PR:
- adds Sentry for error reporting
- adds the contigs listing filters to Analysis -> Contig Viewer. Various prop tweaks to UI components to support this.
- Introduced `featureFlags` to config, so we can toggle features like the Contig GFF Comparison per environment
- adds support for array parameters to be sent to API as repeated URL parameters. This seems bizarre, but is apparently fairly standard in php and apparently Django (who knew). We use this for the Contigs endpoint currently. So e.g. `flag: ['foo', 'bar']` is encoded as `?flag[]=foo&flag[]=bar`. To trigger it the param key must have `[]` in it (to separate this approach from other possible array encoding approaches). So the example would actually by `parameters={'flag[]': ['foo', 'bar']}`
- updates npm lockfile to v2. It *should* be backwards compatible.

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -6,5 +6,9 @@
   "basename": "/metagenomics",
   "enaURL": "https://www.ebi.ac.uk/ena/browser/view/",
   "hmmer": "https://wwwdev.ebi.ac.uk/metagenomics/sequence-search/search/phmmer",
-  "googleMapsKey": "XX"
+  "googleMapsKey": "XX",
+  "sentryDsn": "https://7cb28af2cf254e70ae5eb72733f745e3@o1128655.ingest.sentry.io/6171583",
```
Comment (gustavo-salazar) [inline config.json:10]: I don't know if you have notice that you can have a private config file that it's not pushed to git `config.private.json`
As I'm not familiar with sentry, I'm not sure if this URL should be on the private or if is OK just to be here

Comment (SandyRogers) [reply config.json:10]: Thanks. Yeh, I noticed that. But no, the Sentry key is not secret (it is present in the client code so served to any user of the website). We will potentially have different keys for each environment, but I assume we will do that by replacing this config.json during deployments, so this is just the default.

File: src/components/Analysis/ContigViewer/Table/index.tsx [hunk 2]
```diff
@@ -130,21 +129,20 @@ const ContigsTable: React.FC = () => {
         cols={contigsColumns}
         data={data}
         showPagination={false}
-        title={() => (
-          <>
-            Assembly Contigs
-            <span className="mg-number">{data.meta.pagination.count}</span>
-          </>
-        )}
+        title={() => <>Assembly Contigs ({data.meta.pagination.count})</>}
```
[the remark this answers is not in the document]
Comment (SandyRogers) [reply src/components/Analysis/ContigViewer/Table/index.tsx]: Thanks!

File: src/components/Analysis/ContigViewer/Filter/ContigLength/index.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState, useContext, memo } from 'react';
+
+import Slider from 'components/UI/Slider';
+import Switch from 'components/UI/Switch';
+import ContigsQueryContext from 'components/Analysis/ContigViewer/ContigsQueryContext';
+
+const MIN = 500;
+const MAX = 100000;
```
Comment (gustavo-salazar) [inline src/components/Analysis/ContigViewer/Filter/ContigLength/index.tsx:8]: Any chance the length of the contigs is bigger than this?

File: config.json (hunk 1 above)
Comment (gustavo-salazar) [reply config.json:10]: Cool!
Yeah, one of the steps in jenkins is creating the `config.private.json` which basically overwrites the values of `config.json`. So yeah if this is a default, then it's good to have it here

**Review by gustavo-salazar — APPROVED** [review]
OK, so happy with the changes.
I think the only issue is the =`TextInputDebouncer` but that bug is not related with this PR so you can merge this, and we can try to fix that issue independently

File: src/components/Analysis/ContigViewer/Filter/ContigLength/index.tsx (hunk 3 above)
Comment (SandyRogers) [reply src/components/Analysis/ContigViewer/Filter/ContigLength/index.tsx]: Turns out, yes! Just pushed support for lengths up to 10E6 (I believe that is longer than what we have). This needed a logarithmic scale on the slider to be usable over the range of expected contig lengths.

**Pull request merged**